site/docs/advanced/reliability.md

```diff
@@ -11,8 +11,134 @@ All errors that should be expected to happen (failing API calls, failing network
 You should make sure to always `await` all promises, or at least call `catch` on them if you ever don't want to `await` stuff.
 Use a linting rule to make sure you cannot forget this.
 
+## Graceful shutdown
+
 For bots that are using long polling, there is one more thing to consider.
-As you are going to stop your instance during operation at some point again, you should consider catching `SIGTERM` and `SIGINT` events, and call `bot.stop` (built-in long polling) or stop your bot via its [handle](https://doc.deno.land/https/deno.land/x/grammy_runner/mod.ts#RunnerHandle) (grammY runner).
+As you are going to stop your instance during operation at some point again,
+you should consider catching `SIGTERM` and `SIGINT` events, and call `bot.stop`
+(built-in long polling) or stop your bot via its
+[handle](https://doc.deno.land/https/deno.land/x/grammy_runner/mod.ts#RunnerHandle)
+(grammY runner):
+
+### Simple long polling
+
+<CodeGroup>
+
+<CodeGroupItem title="TS" active>
+
+```ts
+import { Bot } from "grammy";
+
+const bot = new Bot("<token>");
+
+// Stopping the bot when Node process
+// is about to be terminated
+['SIGTERM', 'SIGINT'].forEach(signal =>
+  process.once(signal, () => bot.stop())
+);
+
+await bot.start();
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="JS">
+
+```js
+const { Bot } = require("grammy");
+
+const bot = new Bot("<token>");
+
+// Stopping the bot when Node process
+// is about to be terminated
+['SIGTERM', 'SIGINT'].forEach(signal =>
+  process.once(signal, () => bot.stop())
+);
+
+await bot.start();
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="Deno">
+
+```ts
+import { Bot } from "https://deno.land/x/grammy/mod.ts";
+
+const bot = new Bot("<token>");
+
+// Stopping the bot when Node process
+// is about to be terminated
+['SIGTERM', 'SIGINT'].forEach(signal =>
+  process.once(signal, () => bot.stop())
+);
```

Comment:
That's not how you can catch OS signals with Deno. Use this:
```suggestion
// Stopping the bot when Deno process
// is about to be terminated
Deno.addSignalListener('SIGTERM', () => bot.stop());
Deno.addSignalListener('SIGINT', () => bot.stop());
```

Reply:
Didn't know that, thanks.